### PR TITLE
Add Python 3.9 support and tests

### PR DESCRIPTION
**Pull request by bmw**
Fixes https://github.com/certbot/certbot/issues/8134.

You can see the tests minus packaging & test farm tests passing at https://dev.azure.com/certbot/certbot/_build/results?buildId=3035&view=results.

### CODE REVIEW

File: tools/dev_constraints.txt [hunk 1]
```diff
@@ -26,9 +26,15 @@ coverage==4.5.4
...
 dns-lexicon==3.3.17
-dnspython==1.15.0
-docker==3.7.2
-docker-compose==1.25.0
+# There is no version of dnspython that works on both Python 2 and Python 3.9.
+# To work around this, we make use of the fact that subject to other
+# constraints, pip will install the newest version of a package while ignoring
+# versions that don't support the version of Python being used. The result of
+# this is dnspython 2.0.0 is installed in Python 3 while dnspython 1.16.0 is
+# installed in Python 2.
```
Comment (adferrand) [inline tools/dev_constraints.txt:34]: Given that we are using a "glitch" of pip here, I think we should add a dedicated issue to pin properly `dnspython` once Python 2 support is removed, in addition with the inline comment.

File: tools/merge_requirements.py [hunk 2]
```diff
@@ -16,17 +17,22 @@ def process_entries(entries):
...
+    :returns: mapping from a project to its version specifier
     :rtype: dict
     """
     data = {}
     for e in entries:
         e = e.strip()
         if e and not e.startswith('#') and not e.startswith('-e'):
-            project, version = e.split('==')
-            if not version:
+            for comparison in ('==', '<=',):
```
Comment (adferrand) [inline tools/merge_requirements.py:27]: Can we add a TODO to remove this code once it is not needed anymore. I do not think it is worth to keep this capability to use `<=` when the ultimate goal is to consume pinned dependencies here.

**Review by adferrand — CHANGES_REQUESTED** [review]
Great! Some discussions about what to do with some features added here in the future, and we are good to go!

File: tools/merge_requirements.py (hunk 2 above)
Comment (bmw) [reply tools/merge_requirements.py:33]: Done

**Review by adferrand — APPROVED** [review]

**Pull request merged**